### PR TITLE
Fix graph nodes hopping around

### LINKED ISSUE #3302: Dependencies graph skips
**Describe the bug**

When opening an application's graph, the visualization is re-drawn differently every couple of seconds.

**Environment**

- Weave-Gitops Version: 0.15.0

**To Reproduce**

1. Go to an application that has some dependencies
1. Open the "Graph" tab
1. Wait for a couple of seconds

**Expected behavior**

The graph doesn't change.

**Actual Behavior**

The graph is re-drawn differently, making it hard to follow a certain path.

**Additional Context (screenshots, logs, etc)**

https://user-images.githubusercontent.com/387444/214308989-77e12e87-772d-45c9-8eff-ad1e99580af8.mp4

### PR DESCRIPTION
**Pull request by jpellizzari**
Closes #3302

Adds some streamlining to the back-end to make results more consistent, as well as some sorting logic on the front-end to avoid dancing nodes. 

Shoutout to @luizbafilho for the help on the request simplification.

### CODE REVIEW

**Comment by joshri** [comment]
I'm still getting some node skippage and eventually end in the same `context deadline exceeded` error
`{"code":2, "message":"Get \"https://10.96.0.1:443/apis/kustomize.toolkit.fluxcd.io/v1beta2/namespaces/flux-system/kustomizations/flux-system?timeout=30s\": context deadline exceeded", "details":[]`

**Comment by jpellizzari** [comment]
> I'm still getting some node skippage and eventually end in the same `context deadline exceeded` error `{"code":2, "message":"Get \"https://10.96.0.1:443/apis/kustomize.toolkit.fluxcd.io/v1beta2/namespaces/flux-system/kustomizations/flux-system?timeout=30s\": context deadline exceeded", "details":[]`

Some skippage will be inevitable unless we go for a more complex graph data model. We can never be sure the server will be consistent.

**Comment by joshri** [comment]
So we don't know why the server is returning a different number of objects each req?

**Comment by jpellizzari** [comment]
> So we don't know why the server is returning a different number of objects each req?

@joshri  The server does a fan-out of one request per namespace per cluster (in the case of Core, always `1`). If one of those requests succeeds on one poll and fails on another, you will get dancing nodes. We added some extra logic to reduce the likelihood and log when it happens, but we can't prevent it totally.

File: core/server/fluxruntime.go [hunk 1]
```diff
@@ -276,30 +275,34 @@ func (cs *coreServer) GetReconciledObjects(ctx context.Context, msg *pb.GetRecon
 							return
 						}
 
+						if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
```
[the remark this answers is not in the document]
Comment (jpellizzari) [reply core/server/fluxruntime.go:278]: @luizbafilho Does that mean we need to do something different on this PR?

**Pull request merged**